src/algod/v2/mod.rs

```diff
@@ -173,7 +175,13 @@ impl Algod {
         self.broadcast_raw_transaction(&bytes.concat()).await
     }
 
-    /// Broadcasts a raw transaction or transaction group to the network.
+    /// Broadcasts raw transactions to the network.
+    ///
+    /// When passing multiple transactions, the transactions are atomic if they share a [group](algonaut_transaction::transaction::Transaction::group)
+    ///
+    /// Use this when using a third party (e.g. KMD) that delivers directly the serialized signed transaction.
+    ///
+    /// Otherwise, prefer [Self::broadcast_signed_transaction] or [Self::broadcast_signed_transactions]
     pub async fn broadcast_raw_transaction(
```

Comment:
As mentioned, from a "good practices" perspective, we probably should make this function private. `SignedTransaction` can always be constructed by the user, e.g. by deserializing the byte array returned by KMD. Deserializing to serialize to the same format directly after is a bit wasteful, but it's likely negligible (not sure about exact use cases) and KMD doesn't seem to be used very frequently.

For now letting it public, as we might be missing something. Will revisit.